compiler/rustc_resolve/src/ident.rs

```diff
@@ -977,19 +977,18 @@ impl<'ra, 'tcx> Resolver<'ra, 'tcx> {
             let Some(module) = single_import.imported_module.get() else {
                 return Err((Undetermined, Weak::No));
             };
-            let ImportKind::Single { source, target, target_bindings, .. } = &single_import.kind
-            else {
+            let ImportKind::Single { source, target, bindings, .. } = &single_import.kind else {
                 unreachable!();
             };
             if source != target {
                 // This branch allows the binding to be defined or updated later if the target name
                 // can hide the source.
-                if target_bindings.iter().all(|binding| binding.get().is_none()) {
+                if bindings.iter().all(|binding| binding.get().is_pending()) {
```

Comment:
This is not an equivalent change.
`target_binding.is_none()` is equivalent to `matches!(binding, Pending | Ready(None))`, not pending.
I've seen several instances of this mistake in the patch.